src/main/java/org/folio/oaipmh/helpers/AbstractGetRecordsHelper.java

```diff
@@ -484,6 +486,22 @@ protected Future<Response> processRecords(Context ctx, Request request,
     return oaiResponsePromise.future();
   }
 
+  boolean noRecordsFoundResultCheck(Map<String, RecordType> recordsMap, JsonArray items,  VerbType verb){
+    return (recordsMap.isEmpty() && jsonArrayIsEmpty(items)) ||
+      (recordsMap.isEmpty() && jsonArrayNotEmpty(items) && VerbType.GET_RECORD != verb);
```

Comment:
To simplify the expression, I would suggest to use the following:
    `return recordsMap.isEmpty() && (jsonArrayIsEmpty(items) || jsonArrayNotEmpty(items) && VerbType.GET_RECORD != verb);`

Reply:
done